ddl/ddl_api.go

```diff
@@ -3361,10 +3361,11 @@ func CheckModifyTypeCompatible(origin *types.FieldType, to *types.FieldType) (al
 			mysql.TypeBlob, mysql.TypeTinyBlob, mysql.TypeMediumBlob, mysql.TypeLongBlob:
 			skipSignCheck = true
 			skipLenCheck = true
-		case mysql.TypeEnum, mysql.TypeSet:
-			return unsupportedMsg, errUnsupportedModifyColumn.GenWithStackByArgs(unsupportedMsg)
-		default:
+		case mysql.TypeBit:
+			// todo: currently string data type cast to bit are not compatible with mysql, should fix here after compatible
```

Comment:
Do you mean the bit length is required more to change in TiDB?

Reply:
Yes, the Datum.convertToMysqlBit() will treat "1" as ascii 0x31, then lead to data truncate error. But MySQL can cast "1" to 0x01. 

Maybe it's better to create an individual issue to solve it.

Reply:
Great

Reply:
Got it.